metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -91,11 +91,20 @@ public EbeanEntityService(
     return urnToAspects;
   }
 
+  /*
+   * When a user tries to fetch a negative version, we want to index most recent to least recent snapshots.
+   * To do this, we want to fetch the maximum version and subtract the negative version from that. Since -1 represents
+   * the maximum version, we need to add 1 to the final result.
+   */
+  private long calculateVersionNumber(@Nonnull final Urn urn, @Nonnull final String aspectName, @Nonnull long version) {
```

Comment:
Add a preconditions.assert that version is indeed negative when passed into this method?